provider/testdata/data/baked/collator/dia_v1.rs

```diff
@@ -0,0 +1,68 @@
+// @generated
+use icu_provider::prelude::*;
+impl ResourceProvider<::icu_collator::provider::CollationDiacriticsV1Marker>
+    for super::super::BakedDataProvider
+{
+    fn load_resource(
+        &self,
+        req: &DataRequest,
+    ) -> Result<DataResponse<::icu_collator::provider::CollationDiacriticsV1Marker>, DataError>
+    {
+        static VALUES: &[(&str, DataStruct)] = &[("und", UND)];
+        #[allow(clippy::unwrap_used)]
+        let value = VALUES
+            .binary_search_by(|(k, _)| req.options.cmp_bytes(k.as_bytes()).reverse())
+            .map(|i| VALUES.get(i).unwrap().1)
+            .map_err(|_| {
+                DataErrorKind::MissingResourceOptions.with_req(
+                    <::icu_collator::provider::CollationDiacriticsV1Marker>::KEY,
+                    req,
+                )
+            })?;
+        Ok(DataResponse {
+            metadata: DataResponseMetadata::default(),
+            payload: Some(DataPayload::from_owned(zerofrom::ZeroFrom::zero_from(
+                value,
+            ))),
+        })
+    }
+}
+type DataStruct =
+    &'static <::icu_collator::provider::CollationDiacriticsV1Marker as DataMarker>::Yokeable;
+static UND: DataStruct = &::icu_collator::provider::CollationDiacriticsV1 {
+    ce32s: unsafe {
+        static DATA: &[u8] = &[
```

Comment:
I think this would best be discussed on #1926. Having unaligned loads on crabbaked ZeroVecs is not ideal, but it's the obvious implementation. Maybe in the future we can add a `BorrowedStatic` enum version that contains a `&[T]`, but we have to consider the trade-off between code size and runtime gains.